pkg/util/pod/pod.go

```diff
@@ -56,3 +62,36 @@ func gateIndex(p *corev1.Pod, gateName string) int {
 		return g.Name == gateName
 	})
 }
+
+func ReadUIntFromLabel(pod *corev1.Pod, labelKey string) (*int, error) {
+	return ReadUIntFromLabelWithMinMax(pod, labelKey, math.MaxInt)
+}
+
+func ReadUIntFromLabelWithMinMax(obj client.Object, labelKey string, max int) (*int, error) {
```

Comment:
Could we align the name of the function with the parameters? Should there be `min` parameter?

Reply:
Done